File: app/src/main/java/com/example/jetpacktest/main/MainActivity.java
package com.example.jetpacktest.main;

import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.example.jetpacktest.R;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";
    
    Button addScoreButton, resetScoreButton;
    TextView scoreTextView;

//    Integer score = 0;
    ScoreViewModel scoreViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        
        scoreTextView = findViewById(R.id.scoreTextView);
        addScoreButton = findViewById(R.id.addScoreButton);
        resetScoreButton = findViewById(R.id.resetScoreButton);

        scoreViewModel = ViewModelProviders.of(this).get(ScoreViewModel.class);

        scoreViewModel.getScore().observe(this, new Observer<Integer>() {
            @Override
            public void onChanged(@Nullable Integer integer) {
                Log.i(TAG, "onChanged: ");
                scoreTextView.setText(String.valueOf(integer));
            }
        });

        //addScore button listener
        addScoreButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                addScore();
            }
        });
        //resetScore button listener
        resetScoreButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                resetScore();
            }
        });
    }

    public void addScore() {
        scoreViewModel.addScore();
    }
    
    public void resetScore() {
        scoreViewModel.resetScore();
    }
    
    
    
    
    
}

File: app/src/main/java/com/example/jetpacktest/main/MainActivity.kt
package com.example.jetpacktest.main


import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import com.example.jetpacktest.R
import com.example.jetpacktest.livedata.LiveDataActivity
import com.example.jetpacktest.nolivedata.NoLiveDataActivity
import com.example.jetpacktest.noviewmodel.NoViewModelActivity
import com.example.jetpacktest.viewmodel.ViewModelActivity
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setSupportActionBar(mainToolbar)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu to use in the action bar
        val inflater = menuInflater
        inflater.inflate(R.menu.menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle presses on the action bar menu items
        when (item.itemId) {
            R.id.action_nvm -> {
                goToNoViewModelActivity()
                return true
            }
            R.id.action_vm -> {
                goToViewModelActivity()
                return true
            }
            R.id.action_live_data -> {
                goToNoLiveActivity()
                return true
            }
            R.id.action_data_binding -> {
                goToLiveActivity()
                return true
            }
            R.id.action_room -> {

                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

    fun goToNoViewModelActivity() {
        val intent = NoViewModelActivity.newIntent(this)
        startActivity(intent)
    }

    fun goToViewModelActivity() {
        val intent = ViewModelActivity.newIntent(this)
        startActivity(intent)
    }

    fun goToNoLiveActivity() {
        val intent = NoLiveDataActivity.newIntent(this)
        startActivity(intent)
    }

    fun goToLiveActivity() {
        val intent = LiveDataActivity.newIntent(this)
        startActivity(intent)
    }

}



